pkg/apis/rollouts/v1alpha1/types.go

```diff
@@ -284,6 +286,14 @@ type CanaryStrategy struct {
 	ScaleDownDelayRevisionLimit *int32 `json:"scaleDownDelayRevisionLimit,omitempty" protobuf:"varint,12,opt,name=scaleDownDelayRevisionLimit"`
 }
 
+// Analysis configuration for the analysis runs to retain
+type Analysis struct {
+	// SucceedRunHistoryLimit limits the number of old analysis runs succeeded to be retained in a history
+	SucceedRunHistoryLimit *int32 `json:"succeedRunHistoryLimit,omitempty" protobuf:"varint,1,opt,name=succeedRunHistoryLimit"`
+	// FailedRunHistoryLimit limits the number of old analysis runs failed to be retained in a history
+	FailedRunHistoryLimit *int32 `json:"failedRunHistoryLimit,omitempty" protobuf:"varint,2,opt,name=failedRunHistoryLimit"`
```

Comment:
Could you add the field descriptions to the spec doc: https://github.com/argoproj/argo-rollouts/blob/master/docs/features/specification.md

Reply:
Done